test/integration/processing.test.js

```diff
@@ -130,139 +165,334 @@ describe('process batch files', () => {
   })
+
+  test('calls PublishEvent.sendEvent once when an invalid batch header payment amount to header payment amount file is given', async () => {
+    const blockBlobClient = container.getBlockBlobClient(`${storageConfig.inboundFolder}/${TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_HEADER_PAYMENT_AMOUNT_FILE_SFI_PILOT}`)
+    await blockBlobClient.uploadFile(TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_HEADER_PAYMENT_AMOUNT_FILEPATH_SFI_PILOT)
+
+    await pollInbound()
+
+    expect(mockSendEvent).toHaveBeenCalledTimes(1)
+  })
+
+  test('calls PublishEvent.sendEvent with event.name "batch-processing-quarantine-error" when an invalid batch header payment amount to header payment amount file is given', async () => {
+    const blockBlobClient = container.getBlockBlobClient(`${storageConfig.inboundFolder}/${TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_HEADER_PAYMENT_AMOUNT_FILE_SFI_PILOT}`)
+    await blockBlobClient.uploadFile(TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_HEADER_PAYMENT_AMOUNT_FILEPATH_SFI_PILOT)
+
+    await pollInbound()
+
+    expect(mockSendEvent.mock.calls[0][0].name).toBe('batch-processing-quarantine-error')
+  })
+
+  test('calls PublishEvent.sendEvent with event.properties.status "error" when an invalid batch header payment amount to header payment amount file is given', async () => {
+    const blockBlobClient = container.getBlockBlobClient(`${storageConfig.inboundFolder}/${TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_HEADER_PAYMENT_AMOUNT_FILE_SFI_PILOT}`)
+    await blockBlobClient.uploadFile(TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_HEADER_PAYMENT_AMOUNT_FILEPATH_SFI_PILOT)
+
+    await pollInbound()
+
+    expect(mockSendEvent.mock.calls[0][0].properties.status).toBe('error')
+  })
+
+  test('calls PublishEventBatch.sendEvents when an invalid batch header payment amount to invoice lines payment amount file is given', async () => {
+    const blockBlobClient = container.getBlockBlobClient(`${storageConfig.inboundFolder}/${TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_INVOICE_LINES__PAYMENT_AMOUNT_FILE_SFI_PILOT}`)
+    await blockBlobClient.uploadFile(TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_INVOICE_LINES_PAYMENT_AMOUNT_FILEPATH_SFI_PILOT)
+
+    await pollInbound()
+
+    expect(mockSendEventBatch).toHaveBeenCalledTimes(1)
+  })
+
+  test('calls PublishEventBatch.sendEvents with event.name "batch-processing-payment-request-invalid" when an invalid batch header payment amount to header payment amount file is given', async () => {
+    const blockBlobClient = container.getBlockBlobClient(`${storageConfig.inboundFolder}/${TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_INVOICE_LINES__PAYMENT_AMOUNT_FILE_SFI_PILOT}`)
+    await blockBlobClient.uploadFile(TEST_INVALID_BATCH_HEADER_PAYMENT_AMOUNT_TO_INVOICE_LINES_PAYMENT_AMOUNT_FILEPATH_SFI_PILOT)
+
+    await pollInbound()
+
+    expect(mockSendEventBatch.mock.calls[0][0][0].name).toBe('batch-processing-payment-request-invalid')
```

Comment:
@simonnineteen90 

I wonder if there's any `jest` magical `expect` functions that can allow us to make this more readable?

Sommit like `expect(mockSendEventBatch).firstCall()[0].name.toBe('batch-processing-payment-request-invalid')`

Note: we'd still need the [0] index as it's an array of 2 <-- maybe worth making this clearer?

Reply:
Could not find a `.firstCall()` so closest i could get was using this:
`mockSendEventBatch.mock.lastCall[0][0].name` 
This doesnt seem that different to what we currently have so not sure its easier to read. 
Is there something I have missed?